symengine/ruby/lib/symengine/basic.rb

```diff
@@ -0,0 +1,13 @@
+require 'set'
+
+module SymEngine
+  class Basic
+    def free_symbols
+      pr_free_symbols.to_set
+    end
+
+    def hash
+      [self.to_s].hash
```

Comment:
`ULL2NUM(unsigned long long) - take an unsigned long long int and convert it to a FIXNUM object if it will fit; otherwise, convert to a Bignum object`